### DISCUSSION TITLE
Is it possible to import texture from command line?

### DISCUSSION THREAD

**Discussion by pavelhoral** [nesrak1/UABEA#232] [General]
I like the ability to script replacing assets via command line. Is it possible to achieve something similar with texture import (i.e. replacing data of Texture2D)?

EDIT: I have just noticed draft task "Command line support" on the project's task board. So I guess this is not possible.

**Comment by nesrak1** [comment]
See my message on #230 and all the other issues in [this card](https://github.com/users/nesrak1/projects/1?pane=issue&itemId=20027224).

**Reply by nesrak1** [reply]
To add to that, you can still do texture modding from AssetsTools or UnityPy, you will just have to code it rather than using a CLI.

**Reply by pavelhoral** [reply]
Thank you for those suggestions. I will probably pack my 30-ish textures by hand and hope I don't have to do that more than once for now :). I might check those projects and do something more advanced later (or maybe rewrite some of that stuff in NodeJS as that is scripting language of my choice).

**Reply by nesrak1** [reply]
> pack my 30-ish textures by hand 

Not sure if you were doing this already or not, but if you select multi-select a bunch of Texture2Ds and use plugins "batch import textures" you can import multiple textures at once (any textures not found in the directory you choose will be ignored.)

**Reply by pavelhoral** [reply]
Did not know about that. Will try that tomorrow. Thank you!

**Reply by DraKoNiier** [reply]
> > pack my 30-ish textures by hand
> 
> Not sure if you were doing this already or not, but if you select multi-select a bunch of Texture2Ds and use plugins "batch import textures" you can import multiple textures at once (any textures not found in the directory you choose will be ignored.)

I'm having the same problem as the guy there. I want to import a button texture into a game. Then I extract the texture in PNG, edit it as I want in Photoshop and go back to UABE, selecting the texture file I want to replace and when I go to plugins, a blank window opens with no option to import PNG.

![Captura de tela 2024-07-28 002233](https://github.com/user-attachments/assets/6c546d9f-75ea-431b-8098-95ad430d221d)
